crates/rmf_site_editor/src/site/property_plugin.rs

```diff
@@ -79,104 +148,158 @@ impl<T: Property> LastSetValue<T> {
+                **property_modifier = new_value.clone();
+            } else if let Some(modifier_entity) = modifier_entity {
+                commands
+                    .entity(*modifier_entity)
+                    .insert(Modifier::<T>::new(new_value.clone()));
             } else {
-                // No modifier exists in this tree for this scenario/element pairing
-                // Make sure that a modifier for this property exists in the current scenario tree
-                if let Some(modifier_entity) =
-                    get_modifier.scenarios.get(event.in_scenario).ok().and_then(
-                        |(scenario_modifiers, _)| scenario_modifiers.get(&event.for_element),
-                    )
-                {
-                    commands
-                        .entity(*modifier_entity)
-                        .insert(Modifier::<T>::new(fallback_value));
-                    scenario_state.apply(world);
-                } else {
-                    // Modifier entity does not exist, add one
-                    let root_modifier_entity =
-                        commands.spawn(Modifier::<T>::new(fallback_value)).id();
-                    scenario_state.apply(world);
-                    let mut add_modifier = add_modifier_state.get_mut(world);
-                    add_modifier.write(AddModifier::new_to_root(
-                        event.for_element,
-                        root_modifier_entity,
-                        event.in_scenario,
-                    ));
+                let modifier_entity = commands.spawn(Modifier::<T>::new(new_value.clone())).id();
+                commands.trigger(AddModifier::new(
+                    event.element,
+                    modifier_entity,
+                    event.scenario,
+                ));
+            }
+        }
+        UpdateModifier::<T>::Reset => {
+            // Only process resets if this is not a root scenario
+            if parent_scenario.0.is_some() {
+                if let Some(modifier_entity) = modifier_entity {
+                    commands.entity(*modifier_entity).remove::<Modifier<T>>();
                 }
-                continue;
-            };
-
-        let changed = values
-            .get_mut(world, event.for_element)
-            .is_ok_and(|mut value| {
-                *value = new_value.clone();
-                true
-            });
-        if changed {
-            world
-                .commands()
-                .entity(event.for_element)
-                .insert(LastSetValue::<T>::new(new_value));
+            }
         }
     }
+
+    if event.trigger_update_property {
+        commands.trigger(UpdateProperty::new(event.element, event.scenario));
+    }
 }
 
-fn on_add_property<T: Property, F: QueryFilter + 'static + Send + Sync>(
+/// Updates the current scenario's property values based on changes to the property modifiers
+fn on_update_property<T: Property, E: Element>(
+    trigger: Trigger<UpdateProperty>,
```

Comment:
I think the names of `UpdateProperty` and `UpdateModifier` are too similar.

The docs for `UpdateProperty` say
```rust
/// This event is triggered when changes have been made to an element in the
/// current scenario, and requires the element property value to be updated.
```
which I don't think is accurate anymore since this event seems to be mostly about switching the property to the modifier of a specific scenario. Maybe a name like `UseModifierForScenario` would be more accurate?